src/python/pants/backend/go/goals/test.py

```diff
@@ -288,6 +299,68 @@ def compilation_failure(
             exit_code=0,
         )
 
+    testmain_analysis_input_digest = await Get(
```

Comment:
I don't quite grok why this is necessary now but wasn't before?

Reply:
It wasn't necessary before because the test binary was linked to every SDK package archive. Those package archives are going away so now we need to actually figure out what the test binary's SDK dependencies are.